src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py

```diff
@@ -534,22 +537,30 @@ def test_keyvault_secret_soft_delete(self, resource_group):
         })
         _create_keyvault(self, self.kwargs, additional_args='--enable-soft-delete')
         self.cmd('keyvault show -n {kv}', checks=self.check('properties.enableSoftDelete', True))
-        if self.is_live:
-            time.sleep(20)
-
-        # show deleted
-        self.cmd('keyvault secret set --vault-name {kv} -n {sec} --value ABC123',
-                 checks=self.check('value', 'ABC123'))
-        data = self.cmd('keyvault secret delete --vault-name {kv} -n {sec}').get_output_in_json()
-        if self.is_live:
-            time.sleep(40)
 
-        self.kwargs['secret_id'] = data['id']
-        self.kwargs['secret_recovery_id'] = data['recoveryId']
-        self.cmd('keyvault secret list-deleted --vault-name {kv}', checks=self.check('length(@)', 1))
-        self.cmd('keyvault secret list-deleted --vault-name {kv} --maxresults 10', checks=self.check('length(@)', 1))
-        self.cmd('keyvault secret show-deleted --id {secret_recovery_id}', checks=self.check('id', '{secret_id}'))
-        self.cmd('keyvault secret show-deleted --vault-name {kv} -n {sec}', checks=self.check('id', '{secret_id}'))
+        max_timeout = 100
+        cnt_timeout = 0
+        data = {}
+        while cnt_timeout <= max_timeout:
+            try:
+                # show deleted
+                self.cmd('keyvault secret set --vault-name {kv} -n {sec} --value ABC123',
+                         checks=self.check('value', 'ABC123'))
+                data = self.cmd('keyvault secret delete --vault-name {kv} -n {sec}').get_output_in_json()
+                self.kwargs['secret_id'] = data['id']
+                self.kwargs['secret_recovery_id'] = data['recoveryId']
+                self.cmd('keyvault secret list-deleted --vault-name {kv}', checks=self.check('length(@)', 1))
+                self.cmd('keyvault secret list-deleted --vault-name {kv} --maxresults 10',
+                         checks=self.check('length(@)', 1))
+                self.cmd('keyvault secret show-deleted --id {secret_recovery_id}',
+                         checks=self.check('id', '{secret_id}'))
+                self.cmd('keyvault secret show-deleted --vault-name {kv} -n {sec}',
+                         checks=self.check('id', '{secret_id}'))
+            except:  # pylint: disable=bare-except
+                time.sleep(10)
+                cnt_timeout += 10
```

Comment:
I seems not a timeout. It is a counter.

Reply:
Exactly, bad variable name. I will modify it.